### ISSUE TITLE
Swoole + `defer()`, function already exists.

### ISSUE THREAD

**Issue by tobz-nz** [laravel/octane#948]
### Laravel Version

11.23.4

### PHP Version

8.3.10

### Database Driver & Version

mysql5.7:latest docker

### Description

Laravel Sail `v1.32.0`.

When using the `Cache::flexible()` method I get the following error:

```
defer() expects exactly 1 argument, 2 given at /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/Repository.php:521)
```

I noticed this error only happens in the browser not the CLI - which I usually run form my host, not sail/docker.
So I tried it in the sail container - same error.

So dumping a reflection of the function reveals:

```php
$reflFunc = new ReflectionFunction('defer');
dd($reflFunc);

1 ReflectionFunction {#4 ▼
  +name: "defer"
  returnType: "void"
  parameters: {▼
    $callback: ReflectionParameter {#19 ▼
      +name: "callback"
      position: 0
      typeHint: "callable"
    }
  }
  extra: {▼
    extension: "swoole"
    isInternal: true
  }
}
```

Printing `phpinfo()` shows that Sail has Swoole installed. and I assume from this Swoole adds a global `defer()` function, so the Lavavel helper funciton is never defined - and would error if it did - when Swoole is installed.

### Steps To Reproduce

Create a new Laravel 11.23+ project using Laravel Sail (or install Swoole).

run a controller or tinker:

```php
// call it twice
Cache::flexible('test', [1, 5], fn () => 'test');
Cache::flexible('test', [1, 5], fn () => 'test');
```

**Comment by mihaileu** [comment]
I couldn't reproduce it, I run in a controller with swoole enabled and cache driver array.

**Comment by crynobone** [comment]
![cleanshot_2024-09-13_at_21.38.09.png](https://github.com/user-attachments/assets/2869ff08-ec3e-47c2-b6bf-c8ed5916961a)

This seem related

**Comment by AbdullahALT** [comment]
I ran into the same issue when I tried Cache::flexible in a project using Sail, and sure enough it was using swoole's helper instead of Laravel's 

```php
ReflectionFunction {#1618 // app/Http/Controllers/Lookup/LookupsController.php:36
  +name: "defer"
  returnType: "void"
  parameters: {
    $callback: ReflectionParameter {#1625
      +name: "callback"
      position: 0
      typeHint: "callable"
    }
  }
  extra: {
    extension: "swoole"
    isInternal: true
  }
}
```

my cache driver is redis in case this is related

**Comment by puklipo** [comment]
It is not related to Octane. This occurs in environments where the Swoole extension is installed.

There is a conflict between Laravel's global `defer()` and Swoole's global `defer()`.

`defer()` is used in `Cache::flexible()`. Swoole's `defer()` is being used here.
```php
        if (function_exists('defer')) {
            defer($refresh, "illuminate:cache:refresh:{$key}");
        } else {
            $refresh();
        }
```
https://github.com/laravel/framework/blob/da28aabb597ec4ea6f43bcaf82e263a0fad8737f/src/Illuminate/Cache/Repository.php#L520

**Comment by sy-records** [comment]
`defer` is a short named function in swoole that you can turn off via php.ini.

```ini
swoole.use_shortname=off
```

**Comment by tobz-nz** [comment]
Sorry I don't know how I created this on the Octane repo.

I can create a PR on Sail adding this to the all the php.ini's...

> `defer` is a short named function in swoole that you can turn off via php.ini.
> 
> ```ini
> swoole.use_shortname=off
> ```

**Comment by BassemN** [comment]
I believe the `defer` function name is already registered with the `Swoole` extension, which could cause conflicts with existing projects that rely on Swoole’s defer function or have the Swoole extension enabled. Unfortunately, I think Laravel should consider renaming the defer function to avoid these conflicts.

**Comment by taylorotwell** [comment]
I have no plans to rename `defer` at this time. Swoole is used by a very small portion of the Laravel user base so I'd like to find alternative solutions.

**Comment by dash8x** [comment]
Running into this issue on a default Laravel installation on Forge

**Comment by marco-silvestri** [comment]
Running into this issue on an upgraded Laravel 11 project with Sail, where Swoole is clearly installed by default:

/docker/8.3/Dockerfile:33-34
`       && pecl install swoole-5.1.2 \
       && echo "extension=swoole.so" > /etc/php/8.3/cli/conf.d/20-swoole.ini \`

**Comment by AbdullahALT** [comment]
@marco-silvestri 
Swoole is included in sail, I've managed to fix it by doing the following
- create a file in your project `./docker/php/sail-php.ini`
- paste this into the file
```
swoole.use_shortname=off
```
- map it to the container in docker-compose.yml
```
       volumes:
            - '.:/var/www/html'
            - './docker/php/sail-php.ini:/etc/php/8.3/cli/conf.d/99-sail.ini'
```
- shut sail down and run it again, it should be fixed then
```
sail down
sail up -d
```

**Comment by bogusbd** [comment]
Hello, 

I use Forge
Laravel Octane is disabled 


I'm trying to use ``` Cache::flexible() ```

And get this error: 
```
defer() expects exactly 1 argument, 2 given {"exception":"[object] (ArgumentCountError(code: 0): defer() expects exactly 1 argument, 2 given 
```

I assume that Forge pre-installs Swoole (so that it is always ready to be turned on) .... and as a result there is a naming conflict

**Comment by marco-silvestri** [comment]
> @marco-silvestri Swoole is included in sail, I've managed to fix it by doing the following
> 
> * create a file in your project `./docker/php/sail-php.ini`
> * paste this into the file
> 
> ```
> swoole.use_shortname=off
> ```
> 
> * map it to the container in docker-compose.yml
> 
> ```
>        volumes:
>             - '.:/var/www/html'
>             - './docker/php/sail-php.ini:/etc/php/8.3/cli/conf.d/99-sail.ini'
> ```
> 
> * shut sail down and run it again, it should be fixed then
> 
> ```
> sail down
> sail up -d
> ```

It works flawlessly, thanks!

**Comment by Rattone** [comment]
> I have no plans to rename `defer` at this time. Swoole is used by a very small portion of the Laravel user base so I'd like to find alternative solutions.

So why install it with Sail?

**Comment by jordanhavard** [comment]
I also rolled out Cache::flexible updates (hosted on forge not using octane) and ran into the same issue as described. 

Updating the php.ini as per @sy-records fixed the issue after a php-fpm restart.

**Comment by sy-records** [comment]
Fixed via https://github.com/laravel/framework/pull/52801

**Comment by jordanhavard** [comment]
> Fixed via [laravel/framework#52801](https://github.com/laravel/framework/pull/52801)

Wicked 👌 Hopefully will get tagged next week

**Comment by yoeriboven** [comment]
> I have no plans to rename defer at this time. Swoole is used by a very small portion of the Laravel user base so I'd like to find alternative solutions.

I don't use Swoole but the extensions was installed with Forge. My app runs on a new server provisioned by Forge.
